### DISCUSSION TITLE
Ran into a Bug when trying to leave and join a room with the value of None

### DISCUSSION THREAD

**Discussion by Reardon85** [miguelgrinberg/Flask-SocketIO#1994] [Q&A]
Ran into a weird issue. running react front end and I had convoID's initial value set as null.  This worked fine  when I only had the emit for join_convo  but when I added the leave_convo emit as a cleanup act. It caused my backend to stop executing the flask-socketio listeners. I placed loggers in each so I know my useEffect code was working fine and convoId was changing.  I've fixed the issue by setting my initial value of convoId to the  string "null" and it works fine now. So it doesn't  seem to be  async thing. Anyways my code works so I'm fine now.  Just curious if anyone knows about this issue or what it is. 

```javascript
    useEffect(() => {
        socket.emit('join_convo', {  convoId, userId });

        return () => {
            socket.emit('leave_convo', {convoId})  
        }
    }, [convoId]);
```
backend:
```python
@socketio.on('join_convo')
def on_join(data):
    print(f'Joining Convo {data["convoId"]}')
    convo_id = data['convoId']
    join_room(convo_id)
    # emit('user_connected', {'message':  "has entered the room."}, room=convo_id)

@socketio.on('leave_convo')
def on_leave(data):
    print(f'leaving Convo {data["convoId"]}')
    convo_id= data['convoId']
    print(f'leaving Convo {data["convoId"]}')
    leave_room(convo_id)
```

**Comment by miguelgrinberg** [comment]
See the troubleshooting section of the documentation to learn how to enable detailed logging for the server. Then please provide the logs for one of these sessions in which you say the server stops responding to events. Without these details it is hard to know what happened.
